Conjur/conventions/ruby-style-guide.md

```diff
@@ -24,10 +23,163 @@ This guide is intended to be a "living document" in the sense that it's expected
 as we refine and hone our Ruby knowledge base.
 
 ## Style Baseline
-- **Guide to Ruby** ([Why's Poignant Guide](http://poignant.guide/)) - The best place to start for
-  reading about the langauge
-- **bbatsov's style guide** ([Style Guide](https://github.com/bbatsov/ruby-style-guide)) - A
-  community-driven collection of best practices for Ruby development
+- Follow this official guidelines of the Ruby language https://docs.ruby-lang.org/en/master/doc/syntax/methods_rdoc.html
+  
+- Methods whose purpose is to produce a side effect (change a file, make a network call, raise an error, etc) should have verb phrase names. 
+
+  ```ruby
+  def call
+    validate_and_decode_token
+    get_jwt_identity_from_request
+    validate_host_has_access_to_webservice
+    validate_origin
+    validate_restrictions
+    audit_success
+    new_token
+  rescue => e
+    audit_failure(e)
+    raise e
+  end
+  ```
+
+  
+
+- Functions that return an object should have the name of the returned object so these functions should not have verb phrase names.
```

Comment:
"Functions" -- Let's use "Methods" since that is the proper ruby terminology here.

Also add the sentence:

In some cases, these rules may conflict when a side-effectful method also returns a value.   Here the "verb phrase" rule should win since indicating the side effect is more important.